src/senaite/core/exportimport/genericsetup/structure.py

```diff
@@ -483,14 +473,27 @@ def create_or_get(parent, id, uid, portal_type):
     return obj
 
 
+def can_export_type(portal_type):
+    """Returns whether objects from this type can be exported
+    """
+    key = "can-export-%s" % portal_type
+    annotations = IAnnotations(api.get_request())
+    export = annotations.get(key)
+    if export is None:
+        name = "generic_setup_skip_export_types"
+        skip = get_registry_record(name, default=[])
+        annotations[key] = portal_type not in skip
+        export = annotations[key]
+    return export
```

Comment:
I would prefer the following approach:

1. Lookup the skip list from the registry
2. check if the portal_type is in the list and return `True` or `False`
3. wrap it into a [plone.memoize.request](https://github.com/plone/plone.memoize/blob/master/plone/memoize/request.rst) decorator and a cache key function

Or is there a reason you did it custom?

Reply:
I used `plone.memoize` first, but I thought this would be excessive:

```python
def _export_type_cache_key(method, *args):
    return "can-export-%s" % args[0]

@cache(get_key=_export_type_cache_key, get_request='api.get_request')
def can_export_type(portal_type):
    """Returns whether objects from this type can be exported
    """
    name = "generic_setup_skip_export_types"
    skip = get_registry_record(name, default=[])
    return portal_type not in skip
```

So I simply skipped the decorator and relied on `IAnnotations`, that is what memoize's request actually does here: https://github.com/plone/plone.memoize/blob/master/plone/memoize/request.py#L30-L31

Will revert to `plone.memoize.request` decorator though.

Reply:
Well, I think I could use `lambda` instead of this `_export_type_cache_key` function. Let me try

Reply:
Well is even more complex. Since the function belongs to a module, we have to explicitly pass the request to the decorator, and if we use `api.get_request` as I mentioned above, the system complains that "api does not exist".... 

So I have to put `request` as a parameter to both functions: https://github.com/senaite/senaite.core/pull/2516/commits/76c280c8c076e4262c19c62ea91ee403fb4a6a22

Reply:
I think it becomes simpler with `lambda`: https://github.com/senaite/senaite.core/pull/2516/commits/a32cad66c5e4547349aaf4c12b0dd6bef250288e